### PR TITLE
[5879] Option to sort assets by Name/Recently uploaded

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5879

### CODE REVIEW

File: ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx [hunk 1]
```diff
@@ -63,33 +80,151 @@ export function BrowseFilesDialogUI(props: BrowseFilesDialogUIProps) {
...
+                  <Divider orientation="vertical" flexItem className={classes.actionsBarDivider} />
+                  <Button
+                    id="sort-button"
+                    aria-haspopup="true"
+                    aria-expanded={sortMenuOpen ? 'true' : undefined}
+                    onClick={() => setSortMenuOpen(!sortMenuOpen)}
+                    ref={buttonRef}
+                    sx={{ ml: 1, mr: 2 }}
+                  >
+                    <FilterListIcon sx={{ mr: 1 }} />
```
Comment (rart) [inline ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx]: I think buttons have a specific prop for adding icons?

File: ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx [hunk 2]
```diff
@@ -63,33 +80,151 @@ export function BrowseFilesDialogUI(props: BrowseFilesDialogUIProps) {
...
+                    <FormattedMessage id="words.sorting" defaultMessage="Sorting" />
+                  </Button>
+                  <Menu
+                    id="basic-menu"
```
Comment (rart) [inline ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx]: make your own id.
i think the button is missing the aria-controls reference to this menu? look up the right aria thingie

File: ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx [hunk 3]
```diff
@@ -63,33 +80,151 @@ export function BrowseFilesDialogUI(props: BrowseFilesDialogUIProps) {
...
+                          className={classes.sortingSelect}
+                          label={<FormattedMessage id="BrowseFilesDialog.sortBy" defaultMessage="Sort By" />}
+                        >
+                          <MenuItem value={'_score'}>
+                            <FormattedMessage id="words.relevance" defaultMessage="Relevance" />
+                          </MenuItem>
+                          <MenuItem value={'internalName'}>
+                            <FormattedMessage id="words.name" defaultMessage="Name" />
+                          </MenuItem>
+                          {sortKeys.map((name, i) => (
```
Comment (rart) [inline ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx:167]: Formatting looks broken in github. Check you've run prettier and there are no issues in the app build

File: ui/scss/src/forms-engine.scss [hunk 4]
```diff
@@ -210,3 +210,12 @@ h6,
...
 }
+
+// Styles to fix style conflicts between material-ui and bootstrap
+legend[class^="mui-"], hr.MuiDivider-root {
+  width: auto;
+}
+
+.MuiTooltip-popper {
+  z-index: 1500 !important;
+}
```
Comment (rart) [inline ui/scss/src/forms-engine.scss:221]: Glad to see comprehensive testing 👏🏻🙂

File: ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx [hunk 5]
```diff
@@ -63,33 +80,151 @@ export function BrowseFilesDialogUI(props: BrowseFilesDialogUIProps) {
...
+
   return (
     <>
       <DialogBody className={classes.dialogBody}>
-        <Box display="flex">
-          <section className={classes.leftWrapper}>
+        <Box display="flex" className={classes.dialogContent}>
+          <Box className={classes.leftWrapper} display="flex" flexDirection="column" rowGap="20px">
             <FolderBrowserTreeView rootPath={path} onPathSelected={onPathSelected} selectedPath={currentPath} />
-          </section>
```
Comment (rart) [inline ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx]: Doesn't look like we added the title to this column?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx:167]: Does it look correct now? I ran prettier manually but I saw no difference

**Review by rart — APPROVED** [review]

**Pull request merged**